main.go

```diff
@@ -475,7 +476,11 @@ func (c *controller) sync() {
 	statefulsets := make(map[string]*appsv1.StatefulSet)
 	for _, obj := range c.ssetInf.GetStore().List() {
 		sts := obj.(*appsv1.StatefulSet)
-		statefulsets[sts.Name] = sts
+		hashring, ok := sts.Labels[hashringLabelKey]
+		if !ok {
+			continue
+		}
+		statefulsets[hashring] = sts.DeepCopy()
```

Comment:
Discussed offline: as really all we use from this map is the `.ServiceName`, we should just build a map of that instead.